syntax = "proto2";

package com.scalableminds.webknossos.datastore;

import "geometry.proto";

message Segment {
	  required int64 segmentId = 1;
	  optional Vec3IntProto anchorPosition = 2;
	  optional string name = 3;
	  optional int64 creationTime = 4;
    optional ColorProto color = 5;
}

message VolumeTracing {
    enum ElementClass {
        uint8 = 1;
        uint16 = 2;
        uint24 = 3;
        uint32 = 4;
        uint64 = 8;
    }

    optional int64 activeSegmentId = 1;
    required BoundingBoxProto boundingBox = 2;
    required int64 createdTimestamp = 3;
    required string dataSetName = 4;
    required Vec3IntProto editPosition = 5;
    required Vec3DoubleProto editRotation = 6;
    required ElementClass elementClass = 7;
    optional string fallbackLayer = 8;
    optional int64 largestSegmentId = 9;
    required int64 version = 10;
    required double zoomLevel = 11;
    optional BoundingBoxProto userBoundingBox = 12;
    repeated NamedBoundingBoxProto userBoundingBoxes = 13;
    optional string organizationName = 14; // to identify the dataset (may differ from annotation orga)
    repeated Vec3IntProto resolutions = 15;
    repeated Segment segments = 16;
    optional string mappingName = 17; // either a mapping present in the fallback layer, or an editable mapping on the tracingstore
    optional bool mappingIsEditable = 18;
}

message VolumeTracingOpt {
    optional VolumeTracing tracing = 1;
}

message VolumeTracings {
    repeated VolumeTracingOpt tracings = 1;
}
